forc-pkg/src/pkg.rs

```diff
@@ -55,6 +55,7 @@ pub struct Edge {
     /// This may differ from the package name as declared under the dependency package's manifest.
     pub name: String,
     pub kind: DepKind,
+    pub salt: Option<fuel_tx::Salt>,
```

Comment:
Rather than making this an `Option<fuel_tx::Salt>` on the `Edge` type, let's use the type system to guarantee that contract dependencies *always* have a salt, and library dependencies *never* have a salt.

We can do so by moving the `salt` field to the `Contract` variant of the `DepKind`, e.g.

```rust
pub enum DepKind {
    Library,
    Contract {
        salt: fuel_tx::Salt,
    }
}
```